core/cog-view.c

```diff
@@ -74,6 +77,9 @@ cog_view_set_property(GObject *object, unsigned prop_id, const GValue *value, GP
     case PROP_USE_KEY_BINDINGS:
         cog_view_set_use_key_bindings(self, g_value_get_boolean(value));
         break;
+    case PROP_VIEWPORT:
+        cog_view_set_viewport(self, g_value_get_object(value));
+        break;
```

Comment:
I think we don't need this if we make the property read-only, please read below about this.